Content.Shared/Access/Systems/AccessReaderSystem.cs

```diff
@@ -60,20 +63,65 @@ private void OnEmagged(EntityUid uid, AccessReaderComponent reader, ref GotEmagg
         Dirty(reader);
     }
 
+    /// <summary>
+    /// Finds all AccessReaderComponents in the container of the
+    /// required entity.
+    /// </summary>
+    /// <param name="target">The entity to search for a container</param>
+    private List<AccessReaderComponent> FindAccessReadersInContainer(
+        EntityUid target,
+        AccessReaderComponent? accessReader = null)
+    {
+        List<AccessReaderComponent> result = new();
+
+        if (!Resolve(target, ref accessReader))
+            return result;
+
+        if (accessReader.ContainerAccessProvider == null)
+            return result;
+
+        if (!_containerSystem.TryGetContainer(target, accessReader.ContainerAccessProvider, out var container))
+            return result;
+
+        foreach (var entity in container.ContainedEntities)
+        {
+            if (TryComp<AccessReaderComponent>(entity, out var entityAccessReader))
+                result.Append(entityAccessReader);
+        }
+
+        return result;
+    }
+
     /// <summary>
     /// Searches the source for access tags
-    /// then compares it with the targets readers access list to see if it is allowed.
+    /// then compares it with the all targets accesses to see if it is allowed.
     /// </summary>
     /// <param name="source">The entity that wants access.</param>
     /// <param name="target">The entity to search for an access reader</param>
     /// <param name="reader">Optional reader from the target entity</param>
     public bool IsAllowed(EntityUid source, EntityUid target, AccessReaderComponent? reader = null)
     {
-        if (!Resolve(target, ref reader, false))
-            return true;
-        return IsAllowed(source, reader);
-    }
+        List<AccessReaderComponent> accessReaders = new();
+
+        if (Resolve(target, ref reader))
+            accessReaders.Append(reader);
+
+        var ev = new GetRequiredAccessEvent();
+        RaiseLocalEvent(target, ev);
+
+        if (ev.Handled)
+            accessReaders.Append(ev.Access);
+
+        accessReaders.AddRange(FindAccessReadersInContainer(target, reader));
 
+        foreach (var accessReader in accessReaders)
+        {
+            if (IsAllowed(source, accessReader))
+                return true;
+        }
+        return false;
+    }
+    
```

Comment:
I think the new changes can be more separated from the usual behavior of this method (to avoid unforseen behavior, since this is used a lot) like this
```suggestion
            if (!Resolve(target, ref reader, false))
                return true;

            if (FindAccessReadersInContainer(target, reader.Value, out var accessReaderList))
            {
                foreach (var access in accessReaderList)
                {
                    if (IsAllowed(source, access))
                        return true;
                }
            }

            return IsAllowed(source, reader);
        }    
```

Reply:
```C#
if (!Resolve(target, ref reader, false))
    return true;
```
It wouldn't work because now AccessReaderComponent located in door electronics and doors doesn't have it by default (only prototypes with accesses). Besides that everything is good.

Reply:
OK, I already forgot that we get the container name from AccessReaderComponent, so yes, this will work.